Source/WebCore/css/CSSBasicShapes.cpp

```diff
@@ -246,15 +246,13 @@ static bool buildRadii(Vector<String>& radii, const String& topLeftRadius, const
     return radii.size() == 1 && radii[0] == "0px"_s;
 }
 
-static String buildXywhString(const String& insetX, const String& insetY, const String& width, const String& height,
-    const String& topLeftRadiusWidth, const String& topLeftRadiusHeight,
+
+// FIXME: Try to avoid using StringBuilder here.
```

Comment:
Is avoiding it important for some reason?

Reply:
It creates a lot of intermediate strings, but potentially the solution is to make StringBuilder stop doing that rather than avoiding (which is what's been done in many places)

Reply:
It's slightly misleading to say that `StringBuilder` "creates additional strings". It does not do that. It creates a single result string. However, there are two best practices for efficiency:

- When it can be used to build the full result, `makeString` should be more efficient than `StringBuilder` because it computes the length up front and allocates once at the correct size, with no need to grow as we append and truncate when the string is complete
- When putting a string together from serialization of multiple things, using multiple `StringBuilder` objects means creating and destroying multiple buffers. Instead, building it up in a single `StringBuilder` by having functions that append to one avoids allocating and destroying incremental strings. This is not about avoiding `StringBuilder` but about avoiding multiple `StringBuilder` objects.